pkg/kubelet/certificate/kubelet.go

```diff
@@ -94,8 +94,13 @@ func NewKubeletServerCertificateManager(kubeClient clientset.Interface, kubeCfg
 
 	getTemplate := func() *x509.CertificateRequest {
 		hostnames, ips := addressesToHostnamesAndIPs(getAddresses())
-		// don't return a template if we have no addresses to request for
-		if len(hostnames) == 0 && len(ips) == 0 {
+		// don't return a template if we have no addresses to request for.
+		// A Kubernetes Node requires to have at minimum one IP address
+		// because those are used on the Pods field HostIPs and in some cases,
+		// when pods uses hostNetwork: true, as PodIPs.
+		// Nodes that use IP addresses as Hostname are interpreted as an IP
+		// address, so it is possible that are nodes that don't have any DNS name.
+		if len(ips) == 0 {
```

Comment:
I'm not sure about this bit. DNS-only entries *could* be valid

Reply:
it can be valid, but Kubernetes has a strong requirement on have at least one IP, as Pods depend on them, so my thinking is that a Node will need always at least one IP ... hence, any state without IPs  on the node.status.addresses is a transitional state

Reply:
the point of discussion here is when is really valid to request a certificate, do we want to request a certificate for a node that does not have any IP address reported?

Reply:
I think maybe, because reporting ≠ having. Maybe the node is somehow multihomed but any connection based on a resolved IP address will stay up and working through ~magic~ technology.

Reply:
This is a specific problem with external cloud providers that when the initialize the node object the set only the hostname, after that the cloud provider process the node and updates with the IP addresses.
The problem is that , if the node apporver does not consider valid a node with only DNS names and not with IP, there is a impact on node startup because nodes generate a first certificate request that is incomplete